lm_eval/models/mlc_llm.py

```diff
@@ -300,66 +304,152 @@ def create_chat_completion_payload(
             "stop": stop_tokens,
             "top_p": self.top_p,
             "temperature": self.temperature,
+            "loglikelihood": loglikelihood,
         }
 
         return payload
 
-    def prepare_msg(self, request):
+    def prepare_msg(self, request, loglikelihood=False):
         prompt = request[0]
         request_args = request[1]
         until = request_args["until"]
 
-        return self.create_chat_completion_payload(prompt, until)
-
-    def send_request(self, payload):
-        response = requests.post(
-            f"http://{self.ip}:{self.port}{self.url_suffix}", json=payload, headers=self.headers
-        )
-
-        if response.status_code != 200:
-            print(f"Error: {response.status_code} - {response.text}")
-
-        return json.loads(response.text)
+        return self.create_chat_completion_payload(prompt, loglikelihood, until)
+
+    async def send_request(self, payload):
+        success = False
+        async with aiohttp.ClientSession() as session:
+            for _ in range(REPEAT_REQUEST_TO_MLCSERVE_SERVER):
+                async with session.post(
+                    f"http://{self.ip}:{self.port}{self.url_suffix}",
+                    headers=self.headers,
+                    json=payload,
+                ) as response:
+                    response_status = await response.status_code
```

Comment:
Should be `status` instead of `status_code` in `ClientResponse` from aiohttp. See [docs](https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientResponse.status). Same for lines 329 and 334
Also looks like variable `response_status` is defined, but never used.